server/v2/server_test.go

```diff
@@ -75,12 +75,11 @@ func TestServer(t *testing.T) {
 	require.Equal(t, serverCfgs[mockServer.Name()].(*mockServerConfig).MockFieldOne, MockServerDefaultConfig().MockFieldOne)
 
 	// write config
-	err = server.WriteConfig(configPath)
+	tempDir := t.TempDir()
```

Comment:
I think the test makes less sense now because, we read from `testdata` (see `configPath`) and eventually write to a temporary location and read it back and do more assertion.
So testdata would need to be manually updated if there is a change.

Reply:
That was the goal to not write to the testdata folder. Otherwise the file get's added to a PR by mistake easily and causes other tests to fail later due to concurrent access. I have run into this in a PR as well as Marko. The root cause was not easy to find.
The test still confirms that a pre-defined `app.toml` can be imported and that an exported `app.toml` can be imported again. I hope this adds more context and explains my motivation. WDYT?

Reply:
Supersede by https://github.com/cosmos/cosmos-sdk/pull/21645.
The test was just missing a server, so this is why it wasn't rendering the same file.
I still think we should write it to disk.